packages/host/app/services/realm-info-service.ts

```diff
@@ -0,0 +1,56 @@
+import LoaderService from '@cardstack/host/services/loader-service';
+import Service from '@ember/service';
+import { RealmInfo, SupportedMimeType } from '@cardstack/runtime-common';
+import { service } from '@ember/service';
+
+export default class RecentFilesService extends Service {
+  @service declare loaderService: LoaderService;
+  cachedRealmUrlsForFileUrl: Map<string, string> = new Map(); // Has the file url already been resolved to a realm url?
+  cachedRealmInfos: Map<string, RealmInfo> = new Map(); // Has the realm url already been resolved to a realm info?
```

Comment:
how do these cache get cleared for a service?

Reply:
My reasoning for not including cache busting is the assumption that the realm info doesn't change often

Reply:
I guess when the realm icon can actually be updated by the user we can address this? But agreed that we can leave cache-busting to the future.

Reply:
I agree.